File: LeetCode Practice Problems/1161. Maximum Level Sum of a Binary Tree.cpp
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        queue<TreeNode*> q;
        q.push(root);
        int level = 1 , maxi = root->val;
        int cur = 0;
        while(!q.empty())
        {
            queue<TreeNode*> t;
            int s = 0;
            cur++;
            while(!q.empty())
            {
                TreeNode* r = q.front();
                q.pop();
                s+=r->val;
                if(r->left != NULL)
                    t.push(r->left);
                
                if(r->right != NULL)
                    t.push(r->right);
            }
            cout<<cur<<" "<<s<<" "<<maxi<<"\n";
            q = t;
            if(s > maxi)
            {
                level = cur;
                maxi = s;
            }
        }
        
        return level;
    }
};

File: LeetCode/LeetCode Practice Problems/1161-Maximum Level Sum of a Binary Tree.java
class Solution {
    public int maxLevelSum(TreeNode root) {
        Queue<TreeNode> q = new LinkedList();
        if(root != null) {
            q.add(root);
        }
        int level = 0 , ans = 0;
        long maxSum = Long.MIN_VALUE;
        while(!q.isEmpty()) {
            int sz = q.size();
            level++;
            long curSum = 0;
            while(sz > 0) {
                sz--;
                TreeNode t = q.remove();
                curSum += t.val * 1l;
                if(t.left != null) 
                    q.add(t.left);
                if(t.right != null)
                    q.add(t.right);
            }
            if(maxSum < curSum) {
                maxSum = curSum;
                ans = level;
            }
        }
        return ans;
    }
}
